skycoin/skywire
# Generate PubKey from SecKey if only SecKey is set in Visor config

Did you run `make format && make check`? Yes

Fixes #255

Depends on https://github.com/SkycoinPro/skywire-services/pull/128

 Changes:	
- Generate PubKey from SecKey if only SecKey is set in Visor config
- Change `static_secret_key` to `secret_key` in Visor JSON config.
- Change `static_public_key` to `public_key` in Visor JSON config.
- If a necessary config field is not set and default config value is generated, then the value is updated in the JSON config.

How to test this PR:
- Run generic integration env and check if Visor PK is generated correctly for the following conditions:
  - If both PK and SK are set in config, PK from config should be shown.
  - If only SK is set in config, the same PK should be shown.
  - If only PK is set in config, a newly generated PK should be shown.
  - If no keys are set in config, a newly generated PK should be shown.
